libraries/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -422,7 +422,7 @@ static IotMqttError_t _deserializePublish( _mqttConnection_t * pMqttConnection,
     }
 
     /* Free PUBLISH operation on error. */
-    if( status != IOT_MQTT_SUCCESS )
+    if( ( status != IOT_MQTT_NO_MEMORY ) && ( status != IOT_MQTT_SUCCESS ) )
```

Comment:
Is there any other reason that status could have been set to IOT_MQTT_NO_MEMORY? There are other API calls above that return status. It might be safer to instead test for pOperation being NULL.

Reply:
Good point